packages/vscode-extension/src/devices/DeviceManager.ts

```diff
@@ -104,13 +105,15 @@ export class DeviceManager implements DeviceManagerInterface {
       Logger.error("Error fetching emulators", e);
       return [];
     });
-    const simulators =
-      Platform.OS === "macos"
-        ? listSimulators().catch((e) => {
-            Logger.error("Error fetching simulators", e);
-            return [];
-          })
-        : Promise.resolve([]);
+
+    const shouldLoadSimulators = Platform.OS === "macos" && (await checkXcodeExists());
```

Comment:
Maybe we can add additional log here when checkXcodeExists fails. For example "Couldn't list iOS simulators as XCode installation wasn't found" – this will allow us to track issues in case those checks aren't accurate